Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM python:3.9.6-slim-buster as base 
+
+RUN apt-get update \ 
+    && apt-get -y install make build-essential libssl-dev zlib1g-dev \
+    libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+    
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+
+COPY . /app
+
+WORKDIR /app 
+
+ENV PATH "$PATH:/root/.poetry/bin"
+
+RUN poetry install 
```

Comment:
Currently, when you run `docker build` after changing a file, it will go from the `COPY . /app` step, since that will now be different.

That means even if you change index.html, and then rebuild your image, it's going to have to run `poetry install`.

So you can speed up your rebuilds by switching this up a bit:

```
WORKDIR /app
ENV PATH "$PATH:/root/.poetry/bin"

COPY poetry.lock pyproject.toml /app/
RUN poetry install
COPY . /app
```

This way Docker won't bother redoing `poetry install` unless you have actually changed your dependencies.